### DISCUSSION TITLE
LINQ explained with sketches - the eBook

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#17] [General]
# LINQ explained with sketches - the eBook

After covering and collecting multiple parts of LINQ explained with sketches I bundled them all together in a small booklet with more explanation and code samples.
Therefore I present: LINQ explained with sketches - the eBook. Over 30 pages with smaller explanations and more LINQ operations than in my last post. Litte Bonus: There are interactive links with smaller examples to fiddle around with.

https://steven-giesel.com/blogPost/8d12d9ef-c4e6-439c-9f88-46825cf35576

**Comment by manish-are** [comment]
IEnumerable vs IQueryable blog post link pointing to www.apple.com/uk.I guess link is broken

**Reply by linkdotnet** [reply]
Ahhh thanks. I updated the PDF. Thanks for the report @manish-are 
Much appreciated

**Comment by rperezs** [comment]
In Page 7 
When you ask: "So how many even numbers do we have in the enumeration? The answer is: 3." --> The answer is: 2 (2 & 4)
Total numbers in the list are 3.
Also the Console.WriteLine statement has a  missing "{"

**Reply by linkdotnet** [reply]
Thanks for the bug report. I will incorporate that after my holidays and create an updated version.

**Reply by linkdotnet** [reply]
Hey @rperezs thanks again for reporting. I updated the PDF to reflect the newest changes.

**Comment by Bellarmine-Head** [comment]
I love this, and will share with the devs on my team.  Thanks.

One thing though: the second argument is mostly depicted as being much smaller in size than the first argument (where there are two), as if it's of lesser importance.  This is more the case in the PDF version than on this page.  A good example is `Zip` (page 23 in the PDF).  In fact I was constantly having to zoom in / blow up the PDF page just to see what the second argument was.  Also this caused me initial confusion about how the end result was achieved (the result and the first arg being usually of the same size, but the second arg being tiny and hard to read sometimes).  Perhaps all three (first arg on the left, operator + second arg in the middle, end result on the right) could all be depicted at the same size?

**Reply by linkdotnet** [reply]
Hey @Bellarmine-Head. Thanks for the input. It really shouldn't be unreadable. I will have a look into this and will inform you, when I updated the pages and or if I need your input on this one, hope that is fine for you.

Cheers Steven

**Comment by sbenzenko** [comment]
Hello.
On page 7 in code sample, please correct 
`n => n < 10 == 0`
to
`n => n % 2 == 0`

**Reply by linkdotnet** [reply]
Thanks for reporting @sbenzenko. Naming is hard ;) especially if you change that chapter multiple times. Anyway corrected it and uploaded the fixed version.

**Reply by sbenzenko** [reply]
The problem is not in the name, it's in .Where(...). The condition is incorrect:
Currently it's:
var evenNumbers = list.Where(n => **n < 10** == 0);
And it should be:
var evenNumbers = list.Where(n => **n % 2** == 0);

**Reply by linkdotnet** [reply]
I know I know. In its initial version it was `n % 2` and for some reason I changed it. Now I "reverted" it back to `n%2` with a better explanation what's going on.

**Comment by send2vinnie** [comment]
Amazing handbook!

Just found an error in the example of SelectMany.

Instead of:
> new Recipe("Pizza", new() { "Tomato Sauce, Basil" }),

Should be:
> new Recipe("Pizza", new() { "Tomato Sauce", "Basil" }),

**Reply by linkdotnet** [reply]
Oh darn - If I don't forget I will update that issue.

Thanks for reporting!
